xmlsnippets/src/tests.c

```diff
@@ -80,7 +80,9 @@ static gboolean test(gint ordinal, const gchar *input, const gchar *completion_n
 	else if (!completion_needed && !success)
 		return TRUE;
 
-	if (strcmp(c.completion, completion_needed) != 0)
+	if ((!completion_needed && c.completion) ||
+		(completion_needed && !c.completion) ||
+		strcmp(c.completion, completion_needed) != 0)
```

Comment:
Could also use [`g_strcmp0`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strcmp0).

Reply:
Of course, did it and re-pushed.